apps/web/src/data/farcaster/queries/farcasterProfile.ts.ts

```diff
@@ -0,0 +1,23 @@
+import { UserDataType } from '@farcaster/hub-nodejs'
+
+import { farcasterClient } from '../client'
+
+export const getFarcasterProfile = async (fid: number) => {
+  const client = farcasterClient()
+
+  const pfpRes = await client.getUserData({ fid, userDataType: UserDataType.PFP })
+  const nameRes = await client.getUserData({ fid, userDataType: UserDataType.DISPLAY })
+  const fName = await client.getUserData({ fid, userDataType: UserDataType.FNAME })
```

Comment:
can we make these concurrent and wrap in a promise.all?

Reply:
Yep! Done.